be/src/exec/hdfs_scanner.h

```diff
@@ -369,6 +375,10 @@ class HdfsScanner {
     std::shared_ptr<io::CacheInputStream> _cache_input_stream = nullptr;
     std::shared_ptr<io::SharedBufferedInputStream> _shared_buffered_input_stream = nullptr;
     int64_t _total_running_time = 0;
+    std::shared_ptr<HashJoiner> _hash_joiner = nullptr;
```

Comment:
Do you think it's better to let all scanner with same table share the same hash joiner?

Reply:
I think sharing in the same table will lead to incorrect result. In the iceberg scan task, one data file corresponds to multiple delete files. Data files and deleted files in each iceberg scan task cannot be mixed.  it can be shared in the same file. but it is very troublesome to change the current code architecture.